### ISSUE TITLE
Eclipse Che Helm Chart Improvements

### ISSUE THREAD

**Issue by emreyokur** [eclipse/che#21678]
### Summary

Eclipse Che installation process is too complicated especially when deploying it into a restricted environment. Helm is a very handy tool to support different scenarios as you all know, however when I dig into eclipse-che helm chart, it is just a package of separate manifests with no values attributes given to the user to customize the deployment according to the needs. 

I suggest the community to improve the current helm chart to support customizing it without modifying the templates.

**Comment by tolusha** [comment]
Hello @emreyokur 

Thank you for sharing your thoughts. 
Basically all configuration can be done via CheCluster CR [1]
Here is the doc explaining fields [2]

It would be nice to see some contribution! 

[1] https://artifacthub.io/packages/helm/eclipse-che/eclipse-che?modal=crds&file=checlusters.org.eclipse.che
[2] https://www.eclipse.org/che/docs/stable/administration-guide/checluster-custom-resource-fields-reference/

**Comment by emreyokur** [comment]
Hi @tolusha 

I just want to emphasize that current helm chart is just a wrapped form of manifest files. It is currently not properly templated. I guess we are both agreed to this. I had also reviewed the documentation pages you have linked, however, -maybe i am lost at some point- field reference seems to me just explaining the objects that can be customized, but not explaining the details, available configurable settings for each object. 

My aim is to deploy che on a private on-prem Kubernetes cluster which is not open to internet. I managed to build offline devfile and plugin registry images, I pulled all related images to our container registry. Then I go with chectl tool to deploy che and created a patch yaml to override global settings, Unfortunately, tossed to some error every time, I did not quit yet, still striving... Then I noticed the helm chart with a hope to configure values file and reach the solution in an easier way, 

I would love to contribute. You are absolutely right to ask for a contribution. However, in order to create a fully templated helm chart, one should have comprehensive knowledge about all architectural components. Unfortunately, I do not have that much che-expertise yet. :(
